compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -2157,10 +2199,27 @@ impl<'tcx> LifetimeContext<'_, 'tcx> {
         &self,
         err: &mut Diagnostic,
         mut spans_with_counts: Vec<(Span, usize)>,
-        lifetime_names: &FxHashSet<Symbol>,
-        lifetime_spans: Vec<Span>,
-        params: &[ElisionFailureInfo],
+        in_scope_lifetimes: FxIndexSet<LocalDefId>,
+        params: Option<&[ElisionFailureInfo]>,
     ) {
+        let (mut lifetime_names, lifetime_spans): (FxHashSet<_>, Vec<_>) = in_scope_lifetimes
+            .iter()
+            .filter_map(|def_id| {
+                let name = self.tcx.item_name(def_id.to_def_id());
+                let span = self.tcx.def_ident_span(def_id.to_def_id())?;
+                Some((name, span))
```

Comment:
Wouldn't this be the same as getting the ident and passing that through?

Reply:
The point of this commit was to stop storing the ident of parameters in the ribs.  As we only need them in the case of diagnostics, we fetch them at the last moment.
Anyway, this code will go away with https://github.com/rust-lang/rust/pull/97313.